README.md

```diff
@@ -72,19 +72,39 @@ See [action.yml](action.yml) for the full documentation for this action's inputs
+* Do not store credentials in your repository's code.
 * [Grant least privilege](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#grant-least-privilege) to the credentials used in GitHub Actions workflows.  Grant only the permissions required to perform the actions in your GitHub Actions workflows.
-* [Rotate the credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#rotate-credentials) used in GitHub Actions workflows regularly.
 * [Monitor the activity](https://docs.aws.amazon.com/IAM/latest/UserGuide/best-practices.html#keep-a-log) of the credentials used in GitHub Actions workflows.
 
 ## Assuming a Role
-If you would like to use the static credentials you provide to this action to assume a role, you can do so by specifying the role ARN in `role-to-assume`.
-The role credentials will then be configured in the Actions environment instead of the static credentials you have provided.
-The default session duration is 6 hours, but if you would like to adjust this you can pass a duration to `role-duration-seconds`.
+We recommend using GitHub's OIDC provider to get short-lived credentials needed for your actions. 
+Specifying `role-to-assume` without providing an `aws-access-key-id` or a `web-identity-token-file` will signal to the action that you wish to use the OIDC provider.
+The default session duration is 1 hour when using the OIDC provider to directly assume an IAM Role.
+The default session duration is 6 hours when using an IAM User to assume an IAM Role (by providing an `aws-access-key-id`, `aws-secret-access-key`, and a `role-to-assume`) .
+If you would like to adjust this you can pass a duration to `role-duration-seconds`, but the duration cannot exceed the maximum that was defined when the IAM Role was created.
 The default session name is GitHubActions, and you can modify it by specifying the desired name in `role-session-name`.
 
-Example:
+The following table describes which identity is used based on which values are supplied to the Action:
+
+| **Identity Used**                                                | `aws-access-key-id` | `role-to-assume` | `web-identity-token-file` |
+|------------------------------------------------------------------|---------------------|------------------|---------------------------|
+| [✅ Recommended] Assume Role directly using GitHub OIDC provider |                     | ✔                |                           |
+| IAM User                                                         | ✔                   |                  |                           |
+| Assume Role using IAM User credentials                           | ✔                   | ✔                |                           |
+| Assume Role using WebIdentity Token File credentials             |                     | ✔                | ✔                         |
+
+### Examples
+
+```yaml
```

Comment:
I have a general question I've been struggling to find documentation on from Github. What I'm seeing in my test pipelines is we need to set the `id-token: write` permission in the job permissions field in order to be provided the token for OIDC. I can't find this documented by Github anywhere. I'm also finding that when I set that, and thus go down the road of defining the `permissions` block at all, it undoes any other default permissions. This means I need to do something like:

```yaml
permissions:
  id-token: write
  contents: read
```

In order to both get the OIDC token and continue to be able to use the token for normal `actions/checkout@v2` checkouts.

The google auth plugin did seem to allude to this setting when they updated for OIDC I noticed - https://github.com/google-github-actions/auth#usage

I can't find any Github documentation around this, which I assume is because this actual feature is still beta? If AWS or anyone understands how this thing works, maybe it could be documented here :)

Reply:
we're also still waiting on official documenation

Reply:
Awesome, thanks for verifying I'm not just missing something.